### ISSUE TITLE
Baseline is ignored when using the --redact argument

### ISSUE THREAD

**Issue by gps035** [gitleaks/gitleaks#1441]
**Describe the bug**
When using the `--redact` argument, a baseline file is not taken into account, and leaks are still reported.

**To Reproduce**
Starting in a new git repo, commit a file containing a secret. I used a file called `private_key`, with content generated here https://cryptotools.net/rsagen.

```
$ gitleaks detect --redact

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

11:57AM INF 1 commits scanned.
11:57AM INF scan completed in 4.27ms
11:57AM WRN leaks found: 1
```
Leak reported as expected

```
$ gitleaks detect --report-path gitleaks-report.json

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

11:58AM INF 1 commits scanned.
11:58AM INF scan completed in 4.31ms
11:58AM WRN leaks found: 1
$ gitleaks detect --baseline-path gitleaks-report.json

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

11:58AM INF 1 commits scanned.
11:58AM INF scan completed in 4.06ms
11:58AM INF no leaks found
```
Creating a baseline and using it according to instructions in README.md reports no leak.
```
$ gitleaks detect --baseline-path gitleaks-report.json --redact

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

11:58AM INF 1 commits scanned.
11:58AM INF scan completed in 4.05ms
11:58AM WRN leaks found: 1
```
Adding `--redact` argument causes baseline to be ignored

**Expected behaviour**
No leaks should be reported when the `--redact` argument is added

**Basic Info (please complete the following information):**
 - OS:
  ```
  $ hostnamectl | grep Operating
  Operating System: Ubuntu 22.04.4 LTS
  $ hostnamectl | grep Kernel
  Kernel: Linux 6.5.0-41-generic
  ```
 - Gitleaks Version:
  ```
  $ gitleaks version
  8.18.4
  ```

**Additional context**

cc @zricethezav

**Comment by ajpenner** [comment]
If the baseline is created using --report-format set, the baseline is also ignored.

**Comment by fourcube** [comment]
I think the value of the `redact` flag needs to stay the same for the reporting to work correctly, so when creating the baseline, I made sure to also pass the `--redact` flag. 

### Create baseline:
`gitleaks git --report-path gitleaks-report.json --redact=75`

### Perform scan:
`gitleaks git --baseline-path gitleaks-report.json --redact=75`

**Comment by neumantm** [comment]
Yes, I think @fourcube is right. 
However, I would still consider this a bug.
Took me a while to figure this out.
I think the unredacted as well as the redacted version should be checked against the baseline.

**Comment by gps035** [comment]
I agree, I still consider this a bug. I would never expect the baseline creation to be affected by the --redact flag. If I'm creating a baseline, then I'm accepting the risk of any secrets that already exist in the repo, and I wouldn't think much would be gained by redacting them in the baseline file. I still want new secrets to be kept out of CI logs though
